### PR TITLE
Updating supported Docker commands and constraints for 1.1.

### PR DESCRIPTION
**Pull request by stuclem**
Updated [Supported Docker Commands](https://github.com/stuclem/vic-product/blob/dockercommands/docs/user_doc/vic_app_dev/container_operations.md) and [Supported Docker Compose File Options](https://github.com/stuclem/vic-product/blob/dockercommands/docs/user_doc/vic_app_dev/docker_compose_file_options.md) with options and commands that are now supported in 1.1.

- Fixes https://github.com/vmware/vic-product/issues/109
- Fixes https://github.com/vmware/vic-product/issues/47
- Fixes https://github.com/vmware/vic-product/issues/40
- Fixes https://github.com/vmware/vic-product/issues/48
- Fixes https://github.com/vmware/vic-product/issues/42

Started updating the limitations and constraints on using VIC as a developer.

@chengwang86 can you please take a look? Thanks!

### CODE REVIEW

File: docs/user_doc/vic_app_dev/container_operations.md [hunk 1]
```diff
@@ -52,7 +52,7 @@
 |`pause`|[Pause processes in a container](https://docs.docker.com/engine/reference/commandline/pause/)|No|
 |`port`|[Obtain port data](https://docs.docker.com/engine/reference/commandline/port/)|Yes. Displays port mapping data. <br>Supports mapping a random host port to the container when the host port is not specified.|
 |`ps`|[Show running containers](https://docs.docker.com/engine/reference/commandline/ps/)|Yes. Supports the `-a/--all`, `-f/--filter`, `--no-trunc`, and `-q/--quiet` options. Filtering by network name is supported, but filtering by network ID is not supported.|
-|`rename`|[Rename a container](https://docs.docker.com/engine/reference/commandline/rename/)|No|
+|`rename`|[Rename a container](https://docs.docker.com/engine/reference/commandline/rename/)|Yes|
```
Comment (chengwang86) [inline docs/user_doc/vic_app_dev/container_operations.md]: Yes. Name resolution for renamed running containers is not supported.

Comment (chengwang86) [inline docs/user_doc/vic_app_dev/container_operations.md]: Restarting this renamed running container could be a workaround for this issue. But there is no need to mention this workaround here :)

File: docs/user_doc/vic_app_dev/container_operations.md [hunk 2]
```diff
@@ -130,7 +130,7 @@ For information about Docker Compose file support, see [Supported Docker Compose
 | `start`  | [Start services](https://docs.docker.com/compose/reference/start/)  | Yes  |
 | `stop`  | [Stop services](https://docs.docker.com/compose/reference/stop/)  | Yes  |
 | `unpause`  | [Unpause services](https://docs.docker.com/compose/reference/unpause/)  | No. Depends on `docker unpause`.  |
-| `up`  | [Create and start containers]()  | Conditionally supported. Does not work if there are orphaned containers. Depends on `docker rename` and `docker ps --filter`.  |
+| `up`  | [Create and start containers]()  | Conditionally supported. Does not work if there are orphaned containers. Depends on `docker ps --filter`.  |
```
Comment (chengwang86) [inline docs/user_doc/vic_app_dev/container_operations.md]: I think we can say `Yes`. "orphaned containers" is not a problem now that `docker ps --filter` is supported

File: docs/user_doc/vic_app_dev/creating_containerized_app_with_vic.md [hunk 3]
```diff
@@ -3,5 +3,5 @@
 The topics in this section provides guidelines for container developers who want to use vSphere Integrated Containers Engine to develop and deploy a containerized application. They include an example of using [Docker Compose](https://docs.docker.com/compose/) with vSphere Integrated Containers Engine. 
 
 
-- [Constraints of Using vSphere Integrated Containers Engine to Build Applications](constraints_using_vic.md)
+- [Constraints of Using vSphere Integrated Containers Engine with Docker Compose](constraints_using_vic.md)
```
Comment (chengwang86) [inline docs/user_doc/vic_app_dev/creating_containerized_app_with_vic.md]: The link is wrong

File: docs/user_doc/SUMMARY.md [hunk 4]
```diff
@@ -82,7 +82,7 @@
   * [Using vSphere Integrated Containers Engine with vSphere Integrated Containers Registry (Harbor)](vic_app_dev/using_vic_with_harbor.md) 
   * [Container Networking with vSphere Integrated Containers Engine](vic_app_dev/network_use_cases.md)
   * [Creating a Containerized Application](vic_app_dev/creating_containerized_app_with_vic.md)
-    * [Constraints of Using vSphere Integrated Containers Engine to Build Applications](vic_app_dev/constraints_using_vic.md)
+    * [Constraints of Using vSphere Integrated Containers Engine with Docker Compose](vic_app_dev/constraints_using_vic.md)
```
Comment (stuclem) [inline docs/user_doc/SUMMARY.md]: Actually, the topic is correct, it's just badly named. I renamed it to `constraints_using_compose.md`.

File: docs/user_doc/vic_app_dev/build_app_with_vic.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Example of Building an Application with vSphere Integrated Containers Engine #
 
-The example in this topic modifies the [voting app](https://github.com/docker/example-voting-app) by Docker to illustrate how to work around the constraints that the current version of vSphere Integrated Containers Engine imposes. For information about the constraints, see [Constraints of Using vSphere Integrated Containers Engine to Build Applications](constraints_using_vic.md). 
+The example in this topic modifies the [voting app](https://github.com/docker/example-voting-app) by Docker to illustrate how to work around the constraints that the current version of vSphere Integrated Containers Engine imposes. For information about the constraints, see [Constraints of Using vSphere Integrated Containers Engine with Docker Compose](constraints_using_vic.md).
```
Comment (stuclem) [inline docs/user_doc/vic_app_dev/build_app_with_vic.md]: See above.

File: docs/user_doc/vic_app_dev/container_limitations.md [hunk 6]
```diff
@@ -1,41 +1,50 @@
...
 
+## Using `docker-compose` with TLS
+
+vSphere Integrated Containers supports TLS v1.2, so you must configure `docker-compose` to use TLS 1.2. However, `docker-compose` does not allow you to specify the TLS version on the command line. You must use environment variables to set the TLS version for `docker-compose`. For more information, see [`docker-compose` issue 4651](https://github.com/docker/compose/issues/4651). Furthermore, `docker-compose` has a limitation that requires you to set TLS options either by using command line options or by using environment variables. You cannot use a mixture of both command line options and environment variables. 
+
+To use `docker-compose` with vSphere Integrated Containers and TLS, set the following environment variables:<pre>COMPOSE_TLS_VERSION=TLSv1_2
+DOCKER_TLS_VERIFY=1
+DOCKER_CERT_PATH="<i>path to your cert files</i>"</pre>
+
+The certificate file path must lead to `CA.pem`, `client_key.pem`, and `client cert.pem`. You can run `docker-compose` with the following command:<pre>docker-compose -H <i>vch_address</i> -f up</pre>
```
Comment (stuclem) [inline docs/user_doc/vic_app_dev/container_limitations.md]: Removed.

File: docs/user_doc/vic_app_dev/container_operations.md (hunk 1 above)
Comment (stuclem) [inline docs/user_doc/vic_app_dev/container_operations.md]: Actually, I think that it's worth mentioning here. Added "Name resolution for renamed running containers is not supported, but if you restart the container the new name is resolved."

File: docs/user_doc/vic_app_dev/container_operations.md (hunk 2 above)
Comment (stuclem) [inline docs/user_doc/vic_app_dev/container_operations.md]: Done

File: docs/user_doc/vic_app_dev/creating_containerized_app_with_vic.md (hunk 3 above)
Comment (stuclem) [inline docs/user_doc/vic_app_dev/creating_containerized_app_with_vic.md]: See above.

**Comment by stuclem** [comment]
Thanks @chengwang86, all comments incorporated. Can you please take another look and (hopefully) approve?

**Pull request merged**